source/common/router/config_impl.cc

```diff
@@ -664,15 +672,40 @@ std::string RouteEntryImplBase::newPath(const Http::RequestHeaderMap& headers) c
     final_host = processRequestHost(headers, final_scheme, final_port);
   }
 
+  std::string final_path_value;
+  bool path_redirect_has_query = false;
+  /*
+    if `path_redirect` is not empty:
+      if `path_redirect` contains query string:
+        set `final_path` to `path_redirect` and keep query string anyway
+      else if `strip_query`:
+        set `final_path` to `path_redirect` and strip query string
+      else:
+        set `final_path` to `path_redirect` and keep query string if exists
+    else if `strip_query`:
+      set `final_path` to `current_path` and strip query
+    else:
+      set `final_path` to `current_path`
+  */
```

Comment:
Let's remove this comment.

I do think it's worth having a brief note near here that explains that the path_redirect_ query string, if any, takes precedence over the request's query string.